google_guest_agent/windows_accounts.go

```diff
@@ -198,10 +219,40 @@ func createcredsJSON(k windowsKey, pwd string) (*credsJSON, error) {
 	}, nil
 }
 
+func getWinSSHEnabled(md *metadata) bool {
+	var enable bool
+	if md.Project.Attributes.EnableWindowsSSH != nil {
+		enable = *md.Project.Attributes.EnableWindowsSSH
+	}
+	if md.Instance.Attributes.EnableWindowsSSH != nil {
+		enable = *md.Instance.Attributes.EnableWindowsSSH
+	}
+	return enable
+}
+
 type winAccountsMgr struct{}
 
 func (a *winAccountsMgr) diff() bool {
-	return !reflect.DeepEqual(newMetadata.Instance.Attributes.WindowsKeys, oldMetadata.Instance.Attributes.WindowsKeys)
+	oldSSHEnable := getWinSSHEnabled(oldMetadata)
+
+	sshEnable := getWinSSHEnabled(newMetadata)
+	if !reflect.DeepEqual(newMetadata.Instance.Attributes.WindowsKeys, oldMetadata.Instance.Attributes.WindowsKeys) {
+		return true
+	}
+	if !compareStringSlice(newMetadata.Instance.Attributes.SSHKeys, oldMetadata.Instance.Attributes.SSHKeys) {
+		return true
+	}
+	if !compareStringSlice(newMetadata.Project.Attributes.SSHKeys, oldMetadata.Project.Attributes.SSHKeys) {
+		return true
+	}
+	if newMetadata.Instance.Attributes.BlockProjectKeys != oldMetadata.Instance.Attributes.BlockProjectKeys {
+		return true
+	}
+	if sshEnable != oldSSHEnable {
```

Comment:
suggest you move this check first bc it's simplest and saves the scan tests when true

Reply:
Done.